### PR TITLE
Add missing type hints for QDQBertModel

### PR DESCRIPTION
**Pull request by willtai**
# What does this PR do?
Adding missing type hints for QDQBertModel as referenced in this issue.(https://github.com/huggingface/transformers/issues/16059#issuecomment-1160830014).

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@Rocketknight1 
EDIT: Anyone feel free to review too 😄

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Hey, this looks really good! Would you be willing to add type hints to the other model classes in the same file too? (`QDQBertLMHeadModel` and the classes that start with `QDQBertFor...`)

File: src/transformers/models/qdqbert/modeling_qdqbert.py [hunk 1]
```diff
@@ -77,7 +78,7 @@
 ]
 
 
-def load_tf_weights_in_qdqbert(model, config, tf_checkpoint_path):
+def load_tf_weights_in_qdqbert(model, tf_checkpoint_path):
```
Comment (willtai) [inline src/transformers/models/qdqbert/modeling_qdqbert.py:81]: The `config` parameter is not used in this function

**Comment by willtai** [comment]
@Rocketknight1 I am getting the following error for tests

```
E   ImportError: cannot import name 'login' from 'huggingface_hub' (/home/circleci/.local/lib/python3.7/site-packages/huggingface_hub/__init__.py)
```

Is there a dependency for huggingface_hub perhaps?

Also, I have removed the type hints for the config objects passed as `python utils/check_copies.py` fails due to line above `class QDQBertEmbeddings`
```
# Copied from transformers.models.bert.modeling_bert.BertEmbeddings with Bert -> QDQBert
```

**Comment by Rocketknight1** [comment]
@willtai I suspect something like that is the problem, yes. Can you try:

1) Pulling upstream commits from `transformers` to your repository's `main` branch (you can do this in the Github interface)
2) Rebase your PR branch onto `main` locally
3) Force push (`push -f`) your PR branch to Github

This should resolve those issues if they're caused by a recent code change

**Review by Rocketknight1 — APPROVED** [review]

**Comment by Rocketknight1** [comment]
This is perfect, thank you! The rebase looks clean and all tests are passing, so I'm happy to merge it now.

**Pull request merged**